### PR TITLE
Remove getCurrentContext and withSpan from Tracer

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates #1807 

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/trace/TracingContextUtils.java [hunk 1]
```diff
@@ -66,9 +67,46 @@ public static Span getSpanWithoutDefault(Context context) {
...
+   *   Span span = tracer.spanBuilder("my span").startSpan();
+   *   try (Scope ws = TracingContextUtils.currentContextWith(span)) {
+   *     TracingContextUtils.getCurrentSpan().addEvent("my event");
+   *     doSomeOtherWork();  // Here "span" is the current Span.
+   *   }
+   *   span.end();
+   * }
+   * }</pre>
+   *
+   * <p>Prior to Java SE 7, you can use a finally block to ensure that a resource is closed
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/trace/TracingContextUtils.java:85]: Need to remove this, no longer supported :)

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/TracingContextUtils.java]: Done :)

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
LGTM. Now that this will be the official way, I feel like we should do a follow up to rename `TracingContextUtils` to something more friendly ;)

**Comment by Oberon00** [comment]
Shouldn't we just move TracingContextUtils to static Span interface methods? Or even default Span interface methods where it makes sense.

**Comment by jkwatson** [comment]
> Shouldn't we just move TracingContextUtils to static Span interface methods? Or even default Span interface methods where it makes sense.

I think this is a topic of conversation in #1807

**Comment by jkwatson** [comment]
I hate this change, by the way. It makes the Tracer into something silly and nearly useless. It should be called SpanFactory. :(

**Pull request merged**